config/webpack.config.prod.js

```diff
@@ -92,10 +99,53 @@ module.exports = {
     return [autoprefixer];
   },
   plugins: [
+    // Create two common chunks:
+    // - "vendor" chunk is declared in "entry" above.
+    // - "manifest" is a built-in Webpack chunk with its runtime.
+    // This would enable us to enable long-term caching.
+    // The tricky part is that "manifest" will contain the filenames
+    // of the other chunks. This is why we want to embed it into HTML.
+    // Otherwise, changes in any chunk will cause manifest to change,
+    // and thus if manifest isn't separated, its parent will also change.
+    // Relevant discussion: https://github.com/webpack/webpack/issues/1315
+    new webpack.optimize.CommonsChunkPlugin({
+      names: ['vendor', 'manifest']
+    }),
+    // Fixes deterministic hashes so that changes to CSS don't invalidate JS.
+    new WebpackMd5Hash(),
+    // Avoids extra request for the manifest that contains webpack runtime
+    // by embedding it right into HTML as an inline <script>.
+    function embedManifestIntoHTML() {
+      this.plugin('compilation', (compilation) => {
+        compilation.plugin('html-webpack-plugin-before-html-processing', (data, cb) => {
+          var manifestSource;
+          // Find the manifest chunk generated by webpack,
+          // grab its source, and embed it into generated HTML.
+          Object.keys(compilation.assets).forEach(key => {
+            if (key.indexOf('manifest.') !== 0) {
+              return;
+            }
+            var children = compilation.assets[key].children;
+            if (!children || !children[0]) {
+              return;
+            }
+            delete compilation.assets[key];
```

Comment:
Check that your generated HTML doesn't contain a `<script src>` for the manifest - I found I also had to do this to prevent that happening:

https://github.com/szrenwei/inline-manifest-webpack-plugin/blob/45f6927ded903a4bcb9576fe9b8259322de4fc32/index.js#L32-L34